IS05Utils.py

```diff
@@ -413,25 +413,33 @@ def generate_destination_ports(self, port, portId):
             toReturn = []
             try:
                 for entry in constraints:
-                    if "enum" in entry['destination_port']:
-                        values = entry['destination_port']['enum']
-                        toReturn.append(values[randint(0, len(values) - 1)])
-                    else:
-                        if "minimum" in entry['destination_port']:
-                            min = entry['destination_port']['minimum']
-                        else:
-                            min = 5000
-                        if "maximum" in entry['destination_port']:
-                            max = entry['destination_port']['maximum']
+                    #rtp,mqtt
+                    if "destination_port" in entry:
+                        changeable_param = "destination_port"
+                        if "enum" in entry['destination_port']:
+                            values = entry['destination_port']['enum']
+                            toReturn.append(values[randint(0, len(values) - 1)])
                         else:
-                            max = 49151
-                        toReturn.append(randint(min, max))
-                return True, toReturn
-            except TypeError:
-                return False, "Expected a dict to be returned from {}, got a {}: {}".format(url, type(constraints),
-                                                                                            constraints)
+                            if "minimum" in entry['destination_port']:
+                                min = entry['destination_port']['minimum']
+                            else:
+                                min = 5000
+                            if "maximum" in entry['destination_port']:
+                                max = entry['destination_port']['maximum']
+                            else:
+                                max = 49151
+                            toReturn.append(randint(min, max))
+                    #websocket
+                    elif "connection_uri" in entry:
+                        changeable_param = "connection_uri"
+                        toReturn.append("wss://hostwithrandomport:" + repr(randint(5000, 50000)))
```

Comment:
I wonder if some implementations may be unhappy with this, in particular the hostname? It's probably quite tricky to test, but particularly if there's an 'enum' we could only pick values from that, or 'auto/null'.

Reply:
In fact, I think this is flawed - surely `"connection_uri"` is only set by the sender and cannot be changed by the client? In fact none of the three IS-07 parameters are changeable, so the only thing we could do is use `"auto"`.

Reply:
Actually even `"auto"` may not always be valid for `ext_` parameters... See https://github.com/AMWA-TV/nmos-device-connection-management/issues/71.